editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/BeanConfigUtil.java

```diff
@@ -526,7 +527,8 @@ public void editBeanArgument(Element xmlElement, String type, String value) {
 	}
 	
 	public Element createBeanNode(final CamelFile camelFile, String id, String className) {
-		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getPrefix();
+		// get NS prefix from parent document, not route container node
+		final String prefixNS = camelFile.getRouteContainer().getXmlNode().getOwnerDocument().getPrefix();
```

Comment:
So you're proposing we do the full search through the entire namespace mess on the root document note (not the route node) to locate either the spring or blueprint URI and then grab the prefix from that? My worry with that is speed.